### DISCUSSION TITLE
InputNumber: Make behavior of Home/End keys configurable

### DISCUSSION THREAD

**Discussion by codalf** [primefaces/community#1853] [PrimeReact]
I find the behavior of Home/End keys in InputNumber unusual: When min/max is set, Home will set the value to min and End will set it to max. When min/max are not set, these keys have no function.
This is different from standard behavior in an input field, where Home sets the caret to the very left and End to the very right.
Therefore, I would like to see a configuration property, that allows to activate the 'standard behavior'

**Comment by melloware** [comment]
Hmm that seems to be the accessibility guidelines from ARIA specs and what PrimeTek has settled on?  https://primevue.org/inputnumber/#accessibility

**Comment by codalf** [comment]
I think you are referring to the 'ARIA Spinbutton Pattern' (https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/).

At least in the case, where spin buttons are not displayed in InputNumber (showButtons = false, the default), this pattern does not apply.
Even if min/max is supplied, I believe that there are many use cases, where moving the caret with Home/End to thebeginning/end of the field makes more sense and is the expected behavior. Think of entering long numbers or numbers with several decimal places.

**Reply by melloware** [reply]
I can buy that. Please open a GitHub ticket and have all this info in there!
